### PR TITLE
[JENKINS-53721] - Add getter/setter to EnvVars so the envinject plugin can set Platform safely

### PR DESCRIPTION
**Pull request by jeffpearce**
See [JENKINS-53721](https://issues.jenkins-ci.org/browse/JENKINS-53721).

### Proposed changelog entries

* JENKINS-53721, RFE: Developer: Introduce `getPlatform()` and `setPlatform()` methods in `hudson.EnvVars`
* ...

### Submitter checklist

- [X] JIRA issue is well described
- [X] Changelog entry appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
      * Use the `Internal: ` prefix if the change has no user-visible impact (API, test frameworks, etc.)

### Desired reviewers

@mention

### CODE REVIEW

File: core/src/main/java/hudson/EnvVars.java [hunk 1]
```diff
@@ -84,7 +85,23 @@
...
+     * @return The platform.
+     */
+    public @CheckForNull Platform getPlatform() {
+        return platform;
+    }
 
+    /**
+     * Sets the platform for which these env vars target.
+     * @param platform 
+     */
```
Comment (oleg-nenashev) [inline core/src/main/java/hudson/EnvVars.java]: `@since TODO`

File: core/src/main/java/hudson/EnvVars.java [hunk 2]
```diff
@@ -84,7 +85,23 @@
...
+     * @since TODO
+     * @return The platform.
+     */
+    public @CheckForNull Platform getPlatform() {
+        return platform;
+    }
 
+    /**
+     * Sets the platform for which these env vars target.
+     * @param platform
```
Comment (oleg-nenashev) [inline core/src/main/java/hudson/EnvVars.java]: Please add description if you define a parameter

File: core/src/main/java/hudson/EnvVars.java [hunk 3]
```diff
@@ -84,7 +85,24 @@
...
+        return platform;
+    }
 
+    /**
+     * Sets the platform for which these env vars target.
+     * @since TODO
+     * @param platform the platform to set.
+     */
+    public void setPlatform(@Nonnull Platform platform) {
+        this.platform = platform;
```
Comment (daniel-beck) [inline core/src/main/java/hudson/EnvVars.java]: Is this actually something we should be supporting?

**Review by oleg-nenashev — APPROVED** [review]
It makes sense to me. `EnvVars` is a utility class used in plugins, and it's odd to have a field which is initialized only in the `initMaster()` codebase

**Comment by oleg-nenashev** [comment]
There are 3 plugins trying to be smart there: https://github.com/search?q=org%3Ajenkinsci+%22EnvVars.class.get%22+platform&type=Code . According to the code, I see no concern with making it public

SVN Plugin and Surround-SCM use getters only (the last one is copy-paste from SVN IIUC)

**Pull request merged**